src/core/src/energy/functions.rs

```diff
@@ -364,6 +364,50 @@ impl PairPotential for BornMayerHuggins {
     }
 }
 
+/// Morse potential.
+///
+/// The following potential expression is used: `V(r) = De * (1 - exp(-a (r - re))^2 ; where 'r' is the distance between the atoms, 
+/// 're' is the equilibrium bond distance ('r0'), 'De' is the well ('depth'), and 'a' controls the ('width') of the potential
+///
+/// # Examples
+///
+/// ```
+/// use lumol::energy::Potential;
+/// use lumol::energy::MorsePotential;
+///
+/// let potential = MorsePotential{width: 2.0, r0: 1.5, depth: 4.0};
+/// assert_eq!(morse.energy(1.0), 2.703517287822119);
+/// assert_eq!(morse.force(1.0), -37.12187076378477);
```

Comment:
Those tests are the same as the unit tests below, right? They currently fail since *r0* is different than in the unit tests. Maybe a good additional test would be to hand the distance where force and energy are zero?